ios/RNFetchBlobFS.m

```diff
@@ -437,17 +437,31 @@ + (void) readFile:(NSString *)path
         __block NSData * fileContent;
         if(asset != nil)
         {
-          [[PHImageManager defaultManager]
-           requestImageDataForAsset:asset
-           options:nil
-           resultHandler:^(NSData *imageData,
-                           NSString *dataUTI,
-                           UIImageOrientation orientation,
-                           NSDictionary *info)
-           {
-             fileContent = [NSData dataWithData:imageData];
-             [[self class] readFileContent:fileContent encoding:encoding onComplete:onComplete];
-           }];
+            if(asset.mediaType == PHAssetMediaTypeVideo) {
+
+                [[PHImageManager defaultManager]
+                requestAVAssetForVideo:asset
+                options:nil
+                resultHandler:^(AVAsset * _Nullable asset,
+                                AVAudioMix * _Nullable audioMix,
+                                NSDictionary * _Nullable info) {
+                    AVURLAsset* myAsset = (AVURLAsset*)asset;
```

Comment:
U should check if its a `AVURLAsset` instead of just forcefully assign `AVAsset` to `AVURLAsset`.

```
if (![asset isKindOfClass:[AVURLAsset class]]) {// error}
```